lightning/src/ln/channel.rs

```diff
@@ -2953,6 +2960,29 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let total_fee = commit_tx_fee_sat(feerate_per_kw, (inbound_stats.pending_htlcs as u64 + outbound_stats.pending_htlcs as u64));
+		let holder_balance_after_fee_sub = (self.value_to_self_msat / 1000).checked_sub(total_fee).map(|a| a.checked_sub(outbound_stats.pending_htlcs_value_msat / 1000)).unwrap_or(None).unwrap_or(u64::max_value());
```

Comment:
Wouldn't we want to `unwrap_or(0)`, not `max_value`? I believe as is, if `holder_balance` goes negative, we'll update the fee